staff/maryam-malek/js-arrays/fill.js

```diff
@@ -0,0 +1,24 @@
+function fill(arr, value, start, end) {
+    var newArr = [];
+
+    if(end && start){
```

Comment:
puedes ahorrarte mucho código si pones un condicional ternario en el for que revise si existen las condiciones de start y end

for (var i = ((start)? start: 0) ; i < (end? end: arr.length); i++){
        arr[i] = value;
    }
    return arr